src/test/java/MzMLImportTest.java

```diff
@@ -0,0 +1,145 @@
+public class MzMLImportTest {
+
+    private static final Logger logger = Logger.getLogger(MzMLImportTest.class.getName());
+    private static MZmineProject project;
+
+    /**
+     * Init MZmine core in headless mode with the options -r (keep running) and -m (keep in memory)
+     */
+    @BeforeAll
+    public void init() {
+        logger.info("Getting project");
+        project = MZmineCore.getProjectManager().getCurrentProject();
+    }
+
+    @AfterAll
+    public void tearDown() {
+        //clean the project after this integration test
+        MZmineTestUtil.cleanProject();
+    }
+
+
+    @Test
+    @Order(1)
+    @DisplayName("Test data import of mzML and mzXML without advanced parameters")
+    void dataImportTest() throws InterruptedException, URISyntaxException {
+        File[] files = new File[]{
+                new File(MzMLImportTest.class.getClassLoader().getResource("rawdatafiles/DOM_a.mzML").getFile()), //ind 0
+                new File(MzMLImportTest.class.getClassLoader().getResource("rawdatafiles/DOM_b.mzXML").getFile()), //ind 1
+                new File(MzMLImportTest.class.getClassLoader().getResource("rawdatafiles/DOM_a_invalid_header.mzML").getFile()), //ind 2
+                new File(MzMLImportTest.class.getClassLoader().getResource("rawdatafiles/DOM_b_invalid_header.mzXML").getFile()) //ind 3
+        };
+
+        AllSpectralDataImportParameters paramDataImport = new AllSpectralDataImportParameters();
+        paramDataImport.setParameter(AllSpectralDataImportParameters.fileNames, files);
+        paramDataImport.setParameter(SpectralLibraryImportParameters.dataBaseFiles, new File[0]);
+        paramDataImport.setParameter(AllSpectralDataImportParameters.advancedImport, false);
+
+        logger.info("Testing data import of mzML and mzXML without advanced parameters");
+        TaskResult finished = MZmineTestUtil.callModuleWithTimeout(30,
+                AllSpectralDataImportModule.class, paramDataImport);
+
+        // should have finished by now
+        assertEquals(TaskResult.FINISHED, finished, () -> switch (finished) {
+            case TIMEOUT -> "Timeout during data import. Not finished in time.";
+            case ERROR -> "Error during data import.";
+            case FINISHED -> "";
+        });
+
+        assertEquals(4, project.getDataFiles().length);
+
+        //todo check all scans and mass lists
+        for (RawDataFile raw : project.getCurrentRawDataFiles()) {
+            for (Scan scan : raw.getScans()) {
+                assertNotNull(scan);
+//                assertNotNull(scan.getMassList());
+            }
+        }
+
+        //check if number of scans matches in files with/without 3-byte symbols
+        RawDataFile rawA = project.getCurrentRawDataFiles().get(2);
+        RawDataFile rawB = project.getCurrentRawDataFiles().get(1);
+        RawDataFile rawAInvalid = project.getCurrentRawDataFiles().get(3);
+        RawDataFile rawBInvalid = project.getCurrentRawDataFiles().get(0);
```

Comment:
I guess its run on single thread - but as this is not really a requisite maybe better to get the RawDataFiles by name instead of index. This would make it safer also in the future

Reply:
could also just do project.getCurrentRawDataFiles().stream().filter(r -> r.getName().contains("DOM_a").toList()
Same for b and then you have two lists where you can check all things in two loops one for a one for b

Reply:
I retrieve files by name now. With the list I'm not sure if it would be convenient as I want to compare valid file with invalid